components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java

```diff
@@ -2859,41 +2862,118 @@ public static String preprocessUsername(String username, AuthenticationContext c
 
         ResolvedUserResult resolvedUserResult = new ResolvedUserResult(ResolvedUserResult.UserResolvedStatus.FAIL);
         if (FrameworkServiceDataHolder.getInstance().getMultiAttributeLoginService().isEnabled(tenantDomain)) {
-            resolvedUserResult = FrameworkServiceDataHolder.getInstance().getMultiAttributeLoginService().
-                    resolveUser(loginIdentifier, tenantDomain);
+            resolvedUserResult = FrameworkServiceDataHolder.getInstance().getMultiAttributeLoginService()
+                    .resolveUser(loginIdentifier, tenantDomain);
         }
         return resolvedUserResult;
     }
 
     /**
-     * Validate the username when email username is enabled.
+     * Process multi-attribute login identification using the given username.
      *
-     * @param username Username.
-     * @param context Authentication context.
-     * @throws InvalidCredentialsException when username is not an email when email username is enabled.
+     * @param userName The username given for multi-attribute login.
+     * @return resolvedUserResult with SUCCESS status if enable multi attribute login. Otherwise, returns
+     * resolvedUserResult with FAIL status.
      */
-    public static void validateUsername(String username, AuthenticationContext context)
-            throws InvalidCredentialsException {
+    public static ResolvedUserResult processMultiAttributeLoginIdentification(String userName) {
 
-        if (IdentityUtil.isEmailUsernameEnabled()) {
-            String tenantAwareUsername = MultitenantUtils.getTenantAwareUsername(username);
-            if (StringUtils.countMatches(tenantAwareUsername, "@") < 1) {
-                context.setProperty(CONTEXT_PROP_INVALID_EMAIL_USERNAME, true);
-                throw new InvalidCredentialsException("Invalid username. Username has to be an email.");
+        ResolvedUserResult resolvedUserResult = new ResolvedUserResult(ResolvedUserResult.UserResolvedStatus.FAIL);
+        String tenantDomain = MultitenantUtils.getTenantDomain(userName);
+
+        if (isTenantAvailable(tenantDomain)) {
+            resolvedUserResult = processMultiAttributeLoginIdentification(
+                    MultitenantUtils.getTenantAwareUsername(userName), tenantDomain);
+            if (resolvedUserResult != null && ResolvedUserResult.UserResolvedStatus.SUCCESS.equals(
+                    resolvedUserResult.getResolvedStatus())) {
+                return resolvedUserResult;
             }
         }
+
+        // If the email claim is set as a login attribute and the identifier is an email address, try
+        // login as email in super tenant.
+        if (!MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain) && userName.indexOf('@') > -1
+                && isMultiAttributeLoginEnabled(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME)
+                && getMultiAttributeLoginEnabledClaims(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME).contains(
```

Comment:
`getMultiAttributeLoginEnabledClaims(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME).contains...' is this null-safe?